core/src/main/java/org/commonjava/maven/ext/manip/impl/PluginManipulator.java

```diff
@@ -63,6 +65,11 @@
     private Precedence configPrecedence = Precedence.REMOTE;
 
     /**
+     * Used to store mappings of old property to new version.
+     */
+    protected final HashMap<String, String> versionPropertyUpdateMap = new HashMap<String, String>();
```

Comment:
It might be better to have this in PluginState, just to keep the state separated from these components. I don't think is an issue now, but if we ever turn PME into a standalone service that has something like REST endpoints, we won't want to keep state like this in the components. Pushing it into PluginState now avoids a gotcha if that ever comes about.